tests/snap_test.go

```diff
@@ -0,0 +1,201 @@
+package tests
+
+import (
+	"context"
+	"log"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/canonical/matter-snap-testing/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	// enviroment variables
+	specificGpioChip = "GPIO_CHIP"
+	specifGpioLine   = "GPIO_LINE"
+)
+
+var start = time.Now()
+
+const snapName = "matter-pi-gpio-commander"
+const chipToolSnap = "chip-tool"
+
+func TestMain(m *testing.M) {
+	teardown, err := setup() // THIS NEED TO receive "t" but this function doen't have access to it TODO: fix this latter
+	if err != nil {
+		log.Fatalf("Failed to setup tests: %s", err)
+	}
+
+	code := m.Run()
+	teardown()
+
+	os.Exit(code)
+}
+
+const sedCommand = `sed -i 's|/sys/devices/platform/axi/\*.pcie/\*.gpio/gpiochip4/dev|/sys/devices/platform/axi/*.pcie/*.gpio/gpiochip4/dev\n      - /sys/devices/platform/gpio-mockup.*/gpiochip*/dev|' squashfs-root/meta/snap.yaml`
+
+func authorizeGpioMock(path string) (string, error) {
+	_, stderr, err := utils.Exec(nil, "unsquashfs "+path)
+	if err != nil || stderr != "" {
+		log.Printf("stderr: %s", stderr)
+		log.Printf("err: %s", err)
+		return "", err
+	}
+
+	_, stderr, err = utils.Exec(nil, sedCommand)
+	if err != nil || stderr != "" {
+		log.Printf("stderr: %s", stderr)
+		log.Printf("err: %s", err)
+		return "", err
+	}
+
+	directory := filepath.Dir(path)
+	newPath := directory + "/mod_matter-pi-gpio-commander.snap"
+	_, stderr, err = utils.Exec(nil, "mksquashfs squashfs-root "+newPath)
+	if err != nil || stderr != "" {
+		log.Printf("stderr: %s", stderr)
+		log.Printf("err: %s", err)
+		return "", err
+	}
+
+	utils.Exec(nil, "rm -rf squashfs-root")
+	return newPath, nil
+}
+
+func setup() (teardown func(), err error) {
+	var newPath string
+
+	log.Println("[CLEAN]")
+	utils.SnapRemove(nil, snapName)
```

Comment:
`snapName` needs a better name. Without looking on top, it isn't obvious what snap is being removed and installed. 
```suggestion
	utils.SnapRemove(nil, snapName)
```

Reply:
Renamed to `snapMatterPiGPIO`. Thank you!